### DISCUSSION TITLE
InvalidJwtToken custom message/code return

### DISCUSSION THREAD

**Discussion by JonhnyDev** [adonisjs/core#2077] [Help]
Hello can you help me?

I'm trying to return a custom message when the jwt token is invalid / expired but following documents is not working ...

I tried like this:
start / hooks.js:

const { hooks } = require('@adonisjs/ignitor')

hooks.after.providersBooted(() => {
  const Exception = use('Exception')
  Exception.handle('InvalidJwtToken', (error, { response }) => {
      console.log('test')
    return response.send('Bad token')
    //return response.status(200).json({error: 'test'})
  })
})

return:
{
  "error": "E_JWT_TOKEN_EXPIRED: The jwt token has been expired. Generate a new one to continue"
}
status: 500

or

Exceptions/Handler.js:

const BaseExceptionHandler = use('BaseExceptionHandler')

class ExceptionHandler extends BaseExceptionHandler {
  async handle (error, { response, view}) {
    if (error.code === 'E_INVALID_JWT_TOKEN') {
      return response.status(200).json({error: 'test'})
    }
    return super.handle(...arguments)
  }
}

module.exports = ExceptionHandler

return:
{
  "error": "E_JWT_TOKEN_EXPIRED: The jwt token has been expired. Generate a new one to continue"
}
status: 500

**Comment by Mint-Joy** [comment]
Use try, catch, like
on example auth

```
    try {
      user = await auth.attempt(validated.nickname, validated.password)
    } catch (e) {
      if (e.code === 'E_INVALID_AUTH_UID') {
        return response.json({ res: 11, msg: 'User not found' })
      } else if (e.code === 'E_INVALID_AUTH_PASSWORD') {
        return response.json({ res: 12, msg: 'Invalid password' })
      }
    }
```
and everything will work out
